src/connection/crypto.rs

```diff
@@ -6,7 +6,7 @@ use num_traits::{Zero, One};
 use num_bigint::BigUint;
 use ring::aead;
 use ring::aead::{SealingKey, OpeningKey, Algorithm};
-use ring::rand::SystemRandom;
+use ring::rand::{SecureRandom, SystemRandom};
```

Comment:
Now `fill()` is only available through the `SecureRandom` trait so that trait has to be in scope to call it.